src/support/utf8.h

```diff
@@ -22,8 +22,15 @@ size_t u8_wc_toutf8(char *dest, uint32_t ch);
 /* character number to byte offset */
 DLLEXPORT size_t u8_offset(const char *str, size_t charnum);
 
-/* byte offset to character number */
-DLLEXPORT size_t u8_charnum(const char *s, size_t offset);
+/**
+ * @brief      Calculates number of characters in UTF-8 encoded string
+ *
+ * @param[in]  iStr UTF-8 encoded string (or substring)
+ * @param[in]  iLen Length of string (or substring) in bytes
+ *
+ * @return     Number of logical characters (or codepoints) in string (or substring)
+ */
+DLLEXPORT size_t u8_charnum(const char *iStr, size_t iLen);
```

Comment:
You're using different argument names here. Also, I don't think the docs should be repeated in the headers.